File: Cpp/401 - 600/504 Base 7.cpp
// 4ms, 100.0%; 8.3MB, 100.0%
static int x = []() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution
{
public:
	string convertToBase7(int num)
	{
		if(num == 0) return "0";
		string res = "";
		int pos = abs(num);
		while(pos) {
			int tmp = pos / 7;
			res += to_string(pos - 7 * tmp);
			pos = tmp;
		}
		if(num < 0) res += "-";
		return string(res.rbegin(), res.rend());
	}
};

File: Python3/401 - 600/504 Base 7.py
# 40ms, 61.92%; 13.2MB, 5.55%
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0: return '0'
        res, pos = [], abs(num)
        while pos:
            tmp = pos // 7
            res.append(pos - 7 * tmp)
            pos = tmp
        if num < 0: res.append('-')
        return ''.join([str(r) for r in res[::-1]])
